src/app/static/src/app/store/dataExploration/actions.ts

```diff
@@ -0,0 +1,34 @@
+import {LanguageActions} from "../language/language";
+import {ActionTree} from "vuex";
+import {DataExplorationState} from "./dataExploration";
+import {LanguageMutation} from "../language/mutations";
+import {changeLanguage} from "../language/actions";
+
+export const actions: ActionTree<DataExplorationState, DataExplorationState> & LanguageActions<DataExplorationState> = {
+
+    async changeLanguage(context, payload) {
+        const {commit, dispatch, rootState} = context;
+
+        if (rootState.language === payload) {
+            return;
+        }
+
+        commit({type: LanguageMutation.SetUpdatingLanguage, payload: true});
+
+        await changeLanguage<DataExplorationState>(context, payload);
+
+        const actions: Promise<unknown>[] = [];
+
+        if (rootState.baseline?.iso3) {
+            actions.push(dispatch("metadata/getPlottingMetadata", rootState.baseline.iso3));
+        }
+
+        if (Object.keys(rootState.genericChart.datasets).length > 0) {
+            actions.push(dispatch("genericChart/refreshDatasets"));
+        }
+
+        await Promise.all(actions);
+        commit({type: LanguageMutation.SetUpdatingLanguage, payload: false});
```

Comment:
It is a pity to have so much shared code between this and the corresponding action in `root`- I can see that tripping us up if we ever need to change the shared logic. Could have a shared util which each calls, which takes the rootState, commit and dispatch as parameters, along with a flag to indicate whether all root actions need to be done? But maybe that's a bit convoluted.

Reply:
Well, better than duplication, I had prefer to add it to utils.